src/libsystemd/sd-journal/journal-file.c

```diff
@@ -707,7 +707,7 @@ static int journal_file_allocate(JournalFile *f, uint64_t offset, uint64_t size)
         /* Note that the glibc fallocate() fallback is very
            inefficient, hence we try to minimize the allocation area
            as we can. */
-        r = posix_fallocate(f->fd, old_size, new_size - old_size);
+        r = TEMP_FAILURE_RETRY(posix_fallocate(f->fd, old_size, new_size - old_size));
```

Comment:
this doesn't look right. posix_fallocate() is weird and returns the error as return value and doesn't set `errno`. But TEMP_FAILURE_RETRY checks errno.

That said, I'd really like to understand why EINTR should be retruned here. I mean, this typically means a signals was delivered to journald, which we didn't block.  But which signal is that supposed to be? Can you elaborate?

Reply:
At the first glance I couldn't find a reason.
Actually there are appriximately 950 occurences of EINTR in the Linux kernel sources. I.e. the kernel can return EINTR in many cases not actually related to the signals. (That was surprising to me.)
I've looked at the most of them. The most probable ones are related to the power management. Also, as I've written in the bug, at the time of the error happening, I see kernel mesages about CPU suspending/resuming.
So the most probable cause that the syscall is interrupted because the CPU is suspended.
I've tried to write a synthetical test that infinitely creates a new file, the calls posix_fallocate(), then unlinks it. And from another process I've sent infinitely SIGUSR1 (I've implemented an empty handler in the first process.) But none of the syscalls has beed interrupted with EINTR.

Reply:
And you are right about the TEMP_FAILURE_RETRY(). Will change it.

Reply:
Also once I've straced the journald at the moment of the problem. And there was no any recieved signals in the log at that moment. Just fallocate() returned -1 (EINTR).